### ISSUE TITLE
Contract error returning opaque struct as Any to untyped code

### ISSUE THREAD

**Issue by lwhjp** [racket/typed-racket#871]
### What version of Racket are you using?
7.4

### What program did you run?
```
#lang racket

(module t typed/racket
  (struct foo
    ([val : Any])
    #:transparent)
  (struct bar
    ([var : Any]))
  (define (good) : Any
    (foo 'dummy))
  (define (bad) : Any
    (bar 'dummy))
  (provide (all-defined-out)))

(require (submod "." t))
(define x (good))
(define y (bad))
```

### What should have happened?
Program should run without error.

### If you got an error message, please include it here.
```
bad: broke its own contract
  any-wrap/c: Unable to protect opaque value passed as `Any`
  value: #<bar>
  in: the range of
      (-> Any)
  contract from: 
      (/home/leo/code/contract-err.rkt t)
  blaming: (/home/leo/code/contract-err.rkt t)
   (assuming the contract is correct)
  at: /home/leo/code/contract-err.rkt:11.11
```

**Comment by samth** [comment]
This is intentional; there's no way to protect that value, and it could contain arbitrary typed values such as functions.

**Comment by lwhjp** [comment]
It's not clear to me why the contract can be generated for the transparent struct, then - should an error be issued for that case as well?

**Comment by samth** [comment]
No, we can use reflection to generate the contract for transparent structs. You can see the code here: https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/utils/any-wrap.rkt

**Comment by lwhjp** [comment]
Ah, thanks. I haven't looked into this in detail yet, but I wonder if we can use the inspector in this case since the struct is defined in the same module.

In any case, this was a really perplexing error to track down, so it would be nice to give a bit more of a hint as to what's going on either in the error message or the docs.

**Comment by samth** [comment]
If you have a suggestion for an improved error message I'm happy to take the suggestion.
